src/rdf4cpp/writer/BufWriter.hpp

```diff
@@ -315,6 +315,56 @@ struct BufOStreamWriter : BufWriterBase<BufOStreamWriter, OStreamBuffer> {
     }
 };
 
+template<typename OutIter>
+struct OutputIteratorBuffer {
+    OutIter iter;
+    std::array<char, BUFSIZ> buffer_{};
+
+    explicit constexpr OutputIteratorBuffer(OutIter i)
+        : iter(i) {
+    }
+
+    void write_out(char const *end) {
+        char const *b = buffer_.data();
+        while (b != end && b != buffer_.end()) {
```

Comment:
Any particular reason you assume that the output iterator can end?
`std::copy` also assumes that it never ends

Reply:
its the iterator over the buffer